doc-changelog/parse_pr_title.py

```diff
@@ -0,0 +1,345 @@
+    ----------
+    org_name: str
+        The name of the organization.
+    repo_name: str
+        The name of the repository.
+    default_config: bool
+        Whether or not to use the default towncrier configuration for the pyproject.toml file.
+    """
+    # Load pyproject.toml file
+    with open("pyproject.toml", "a+") as file:
+        config = toml.load("pyproject.toml")
+        tool = config.get("tool", "DNE")
+        towncrier = tool.get("towncrier", "DNE")
+
+        # List containing changelog sections under each release
+        changelog_sections = [
+            "added",
+            "dependencies",
+            "documentation",
+            "fixed",
+            "maintenance",
+            "miscellaneous",
+            "test",
+        ]
+
+        # Dictionary containing [tool.towncrier] keys and values
+        towncrier_config_sections = {
+            "directory": '"doc/changelog.d"',
+            "template": '"doc/changelog.d/changelog_template.jinja"',
+            "filename": {"web": '"doc/source/changelog.rst"', "repo": '"CHANGELOG.md"'},
+            "start_string": {
+                "web": '".. towncrier release notes start\\n"',
+                "repo": '"<!-- towncrier release notes start -->\\n"',
+            },
+            "title_format": {
+                "web": f'"`{{version}} <https://github.com/{org_name}/{repo_name}/releases/tag/v{{version}}>`_ - {{project_date}}"',
+                "repo": f'"## [{{version}}](https://github.com/{org_name}/{repo_name}/releases/tag/v{{version}}) - {{project_date}}"',
+            },
+            "issue_format": {
+                "web": f'"`#{{issue}} <https://github.com/{org_name}/{repo_name}/pull/{{issue}}>`_"',
+                "repo": f'"[#{{issue}}](https://github.com/{org_name}/{repo_name}/pull/{{issue}})"',
+            },
+        }
+
+        # Get the package name from [tool.flit.module]
+        flit = tool.get("flit", "DNE")
+        module = name = ""
+        if flit != "DNE":
+            module = flit.get("module", "DNE")
+            if module != ("DNE" or ""):
+                name = module.get("name", "DNE")
+                # If [tool.flit.module] name exists, create the package string
+                if name != ("DNE" and ""):
+                    towncrier_config_sections["package"] = f'"{name}"'
+
+        if default_config:
+            # If there is no towncrier configuration information or if [[tool.towncrier.type]]
+            # is the only towncrier information in the pyproject.toml file
+            if towncrier == "DNE" or len(towncrier) == 1:
+                # Write the [tool.towncrier] section
+                write_towncrier_config_section(file, towncrier_config_sections, True)
+
+        if towncrier != "DNE":
+            # Get the existing [[tool.towncrier.type]] sections
+            types = towncrier.get("type", "DNE")
+            if types != "DNE":
+                remove_existing_types(types, changelog_sections)
```

Comment:
I can only get the [[tool.towncrier.type]] sections if towncrier is not "DNE". Once we know the existing [[tool.towncrier.type]] types, then we can remove them from the changelog_sections list so that we're only appending the missing [[tool.towncrier.type]] sections in line 272